From 6880219ad4ba55ae8a94a34b7a987b3369f7c96f Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Sun, 24 Nov 2024 09:32:29 +0100
Subject: [PATCH] configure: Make sequencer dependent on rawmidi

The sequencer feature requires rawmidi implicitly, and it became more
obvious with UMP support.  Add the dependency check to configure
script.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 1cd22a5984d7..69aeb978af0d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -483,6 +483,12 @@ fi
 AC_SUBST(PYTHON_LIBS)
 AC_SUBST(PYTHON_INCLUDES)
 
+if test "$build_rawmidi" != "yes"; then
+  if test "$build_seq" = "yes"; then
+    AC_ERROR([Cannot enable sequencer without rawmidi])
+  fi
+fi
+
 AM_CONDITIONAL([BUILD_MIXER], [test x$build_mixer = xyes])
 AM_CONDITIONAL([BUILD_PCM], [test x$build_pcm = xyes])
 AM_CONDITIONAL([BUILD_RAWMIDI], [test x$build_rawmidi = xyes])
-- 
2.43.0

